mousebender/simple.py

```diff
@@ -183,14 +200,54 @@ def handle_starttag(self, tag, attrs_list):
         # Links in the simple repository MAY have a data-yanked attribute which
         # may have no value, or may have an arbitrary string as a value.
         yanked = "data-yanked" in attrs, attrs.get("data-yanked") or ""
-
-        self.archive_links.append(
+        self.update_index(
             ArchiveLink(filename, url, requires_python, hash_, gpg_sig, yanked)
         )
 
+    def update_index(self, archive: ArchiveLink):
+        # old way
+        self.archive_links.append(archive)
+
+        # new way, indexed values
+
+        # Dict[str, Dict[str, Tuple[Optional[ArchiveLink], Optional[Dict[Tag, ArchiveLink]]]]]
+        try:
+            proj, ver, build_tag, tags = packaging.utils.parse_wheel_filename(
+                archive.filename
+            )
+
+            if proj not in self.index:
+                self.index[proj] = {}
+            if ver not in self.index[proj]:
+                self.index[proj][ver] = (None, {tags: archive})
+            else:
+                # handle immutability of a Tuple by re-creating it, pull out the old members:
+                sdist_lnk = self.index[proj][ver][0]
+                archive_link_dict = self.index[proj][ver][1] or {}
+                # add the new info:
+                archive_link_dict[tags] = archive
+                # and re-create it:
+                self.index[proj][ver] = (sdist_lnk, archive_link_dict)
+
+        except packaging.utils.InvalidWheelFilename as bad_file_name:
+            # This is either an sdist or something else.
+
+            try:
+                proj, ver = packaging.utils.parse_sdist_filename(archive.filename)
```

Comment:
Not clear if we want to go down this route _yet_. Wheels are what we've concentrated on thus far, but there is an _implication_ that `sdist` entries are interesting after reading some threads on the forums. I'm happy to remove this for the purpose of this issue.

Reply:
sdists will be necessary for pip or any installer that chooses to support PEP 517. They are easy enough to identify (end in `.tar.gz`), so at least collecting them is simple enough.

Reply:
Basically we were ignoring sdists as I didn't want to take on the responsibility of what _was_ an sdist. `packaging.utils.parse_sdist_filename()` takes care of that for us. 😁